sample-compose/data/src/main/java/co/nimblehq/sample/compose/data/extensions/ResponseMapping.kt

```diff
@@ -0,0 +1,56 @@
+package co.nimblehq.sample.compose.data.extensions
+
+import co.nimblehq.sample.compose.data.response.ErrorResponse
+import co.nimblehq.sample.compose.data.response.toModel
+import co.nimblehq.sample.compose.data.service.providers.MoshiBuilderProvider
+import co.nimblehq.sample.compose.domain.exceptions.ApiException
+import co.nimblehq.sample.compose.domain.exceptions.NoConnectivityException
+import com.squareup.moshi.JsonDataException
+import kotlinx.coroutines.flow.FlowCollector
+import kotlinx.coroutines.flow.flow
+import retrofit2.HttpException
+import retrofit2.Response
+import java.io.IOException
+import java.io.InterruptedIOException
+import java.net.UnknownHostException
+import kotlin.experimental.ExperimentalTypeInference
+
+@Suppress("TooGenericExceptionCaught")
+@OptIn(ExperimentalTypeInference::class)
+fun <T> flowTransform(@BuilderInference block: suspend FlowCollector<T>.() -> T) = flow {
+    val result = try {
+        block()
+    } catch (exception: Exception) {
+        throw exception.mapError()
+    }
+    emit(result)
+}
+
+private fun Throwable.mapError(): Throwable {
+    return when (this) {
+        is UnknownHostException,
+        is InterruptedIOException -> NoConnectivityException
+        is HttpException -> {
+            val errorResponse = parseErrorResponse(response())
+            ApiException(
+                errorResponse?.toModel(),
```

Comment:
Then it will become to this:
```suggestion
                errorResponse?.message,
```